ts/screens/messages/MessageRouterScreen.tsx

```diff
@@ -0,0 +1,134 @@
+import { NavigationActions, NavigationInjectedProps } from "react-navigation";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { CreatedMessageWithContentAndAttachments } from "../../../definitions/backend/CreatedMessageWithContentAndAttachments";
+import { LoadingErrorComponent } from "../../features/bonus/bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import { navigateToEuCovidCertificateDetailScreen } from "../../features/euCovidCert/navigation/actions";
+import { EUCovidCertificateAuthCode } from "../../features/euCovidCert/types/EUCovidCertificate";
+import I18n from "../../i18n";
+import { mixpanelTrack } from "../../mixpanel";
+import { loadMessages } from "../../store/actions/messages";
+import { navigateToMessageDetailScreenAction } from "../../store/actions/navigation";
+import { messagesAllIdsSelector } from "../../store/reducers/entities/messages/messagesAllIds";
+import { messageStateByIdSelector } from "../../store/reducers/entities/messages/messagesById";
+import { GlobalState } from "../../store/reducers/types";
+import { InferNavigationParams } from "../../types/react";
+import { isStrictSome } from "../../utils/pot";
+import { MessageDetailScreen } from "./MessageDetailScreen";
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps> &
+  NavigationInjectedProps<InferNavigationParams<typeof MessageDetailScreen>>;
+
+/**
+ * In order to have the final CreatedMessageWithContentAndAttachments, these conditions should be verified:
+ * - props.allMessages should be pot.some
+ * - messageId should be in props.allMessages
+ * - props.messageState(messageId) should be defined and pot.some
+ * @param props
+ */
+const getLoadingState = (
+  props: Props
+): pot.Pot<
+  CreatedMessageWithContentAndAttachments | undefined,
+  string | undefined
+> => {
+  const messageId = props.navigation.getParam("messageId");
+  if (!isStrictSome(props.allMessages)) {
+    void mixpanelTrack("MESSAGE_ROUTING_FAILURE", {
+      reason: "all Messages is not some"
+    });
+    return pot.map(props.allMessages, _ => undefined);
+  }
+  if (!props.allMessages.value.some(id => id === messageId)) {
+    void mixpanelTrack("MESSAGE_ROUTING_FAILURE", {
+      reason: "id not in list"
+    });
+    return pot.noneError(`Cannot found ${messageId} in messages list`);
+  }
+  const messageState = props.messageState(messageId);
+  if (messageState === undefined) {
+    void mixpanelTrack("MESSAGE_ROUTING_FAILURE", {
+      reason: "MessageState is undefined"
+    });
+    return pot.noneError("MessageState is undefined");
+  }
+  return messageState.message;
+};
+
+/**
+ * Choose the screen where to navigate, based on the message.content.eu_covid_cert
+ * @param message
+ * @param props
+ */
+const navigateToScreenHandler = (
+  message: CreatedMessageWithContentAndAttachments,
+  props: Props
+) => {
+  if (message.content.eu_covid_cert !== undefined) {
```

Comment:
```suggestion
  if (euCovidCertificateEnabled && message.content.eu_covid_cert !== undefined) {
```
I think about it with a little delay 😪

Reply:
😱